common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -689,7 +689,7 @@ public int indexOf(CharSequence subString, int start) {
         }
         final byte firstCharAsByte = c2b0(firstChar);
         final int len = offset + start + length - subCount;
-        for (int i = start + offset; i <= len; ++i) {
+        for (int i = start + offset; i <= len && i < length; ++i) {
```

Comment:
I think it will be more correct:
```java
final int len = offset + length - subCount;
```
that is, do not add the `start` to the `len`.

Reply:
sounds good ... let me close this one